id.go

```diff
@@ -1,11 +1,23 @@
 package errorx
 
-import "sync/atomic"
+import (
+	"sync"
+)
 
-var internalID int64
+var internalID uint64 = 0x123456789abcdef0
+var idMtx sync.Mutex
 
 // nextInternalID creates next unique id for errorx entities.
 // All equality comparison should take id into account, lest there be some false positive matches.
-func nextInternalID() int64 {
-	return atomic.AddInt64(&internalID, 1)
+func nextInternalID() uint64 {
```

Comment:
A link to a paper describing these xorshift parameters would look good here. 
As this internal id **must** be unique, it is important to know exactly when to expect the first repetition. 

And since the current version of the code actually never uses the randomness in this id, is this change really needed?

Reply:
No, it isn't needed, and I'm going to revert it.

ps: xorshift is a full period generator for 2^n-1 period.